pkg/replicate/scheme.go

```diff
@@ -246,7 +251,26 @@ func (rs *replicationScheme) ensureBlockIsReplicated(ctx context.Context, id uli
 			// If the origin meta file content and target meta file content is
 			// equal, we know we have already successfully replicated
 			// previously.
-			level.Debug(rs.logger).Log("msg", "skipping block as already replicated", "block_uuid", blockID)
+			if rs.markAfter {
+				deletionMarkFile := path.Join(id.String(), metadata.DeletionMarkFilename)
+				deletionMarkExists, err := rs.fromBkt.Exists(ctx, deletionMarkFile)
+				if err != nil {
+					return errors.Wrapf(err, "check exists %s in bucket", deletionMarkFile)
+				}
+
+				if !deletionMarkExists {
+					level.Debug(rs.logger).Log("msg", "marking block for deletion as already replicated", "block_uuid", blockID)
+					if err := block.MarkForDeletion(ctx, rs.logger, rs.fromBkt, id, "marked for deletion by thanos bucket replicate", promauto.With(nil).NewCounter(prometheus.CounterOpts{})); err != nil {
```

Comment:
Why do we have `promauto.With(nil).NewCounter(prometheus.CounterOpts{}))` here instead of adding something to `replicationScheme.reg`? This would also allow you to write a e2e test that is missing.

Reply:
Thought it is not very informative metric, because we already have `thanos_replicate_blocks_replicated_total` which would be equal to this new metric if we run with `--mark-after`. (Correct me if i'm wrong)